cmd/launcher/flare.go

```diff
@@ -67,23 +53,41 @@ func runFlare(args []string) error {
 	k := knapsack.New(nil, flagController, nil)
 	ctx := context.Background()
 
-	if *flSave == "upload" {
+	type flareDestinationTyp interface {
+		io.WriteCloser
+		Name() string
+	}
+	var flareDest flareDestinationTyp
+	var successMessage string
+
+	switch *flSave {
+	case "upload":
 		shipper, err := shipper.New(k, shipper.WithNote(strings.Join(flagset.Args(), " ")), shipper.WithUploadRequestURL(*flUploadRequestURL))
 		if err != nil {
 			return err
 		}
-		return checkups.RunFlare(ctx, k, shipper, checkups.StandaloneEnviroment)
-	}
+		flareDest = shipper
+		successMessage = "Flare uploaded successfully"
+	case "local":
+		reportName := fmt.Sprintf("kolide_agent_flare_report_%s.zip", ulid.New())
+		reportPath := filepath.Join(*flOutputDir, reportName)
 
-	// saving flare locally
-	reportName := fmt.Sprintf("kolide_agent_flare_report_%s", ulid.New())
-	reportPath := fmt.Sprintf("%s.zip", filepath.Join(*flOutputDir, reportName))
+		flareFile, err := os.Create(reportPath)
+		if err != nil {
+			return fmt.Errorf("creating flare file (%s): %w", reportPath, err)
+		}
+		defer flareFile.Close()
+		flareDest = flareFile
+		successMessage = "Flare saved locally"
+	default:
+		return fmt.Errorf(`invalid save option: %s, expected "local" or "upload"`, *flSave)
 
-	flareFile, err := os.Create(reportPath)
-	if err != nil {
-		return fmt.Errorf("creating flare file (%s): %w", reportPath, err)
 	}
-	defer flareFile.Close()
 
-	return checkups.RunFlare(ctx, k, flareFile, checkups.StandaloneEnviroment)
+	if err := checkups.RunFlare(ctx, k, flareDest, checkups.StandaloneEnviroment); err != nil {
+		return err
+	}
+
+	level.Info(logger).Log("msg", successMessage, "file", flareDest.Name())
```

Comment:
I like how tidy adding `flareDest.Name()` is

Reply:
Tiny inline interface FTW. (Not sure if inline is correct, but 🤷 )